### ISSUE TITLE
PERF403 potential false positive

### ISSUE THREAD

**Issue by LefterisJP** [astral-sh/ruff#7446]
I enabled `--preview` with version `0.0.290` and got an interesting hit for `PERF403`.

Code looks kind of like this:

```python
mapping = {}

animals = ['dog', 'cat', 'bird', 'elephant']
other_mapping = {'snake': 69, 'owl': 42}

for idx, animal in enumerate(animals):
    if animal == 'elephant':
        continue
    mapping[animal] = idx

for animal, idx in other_mapping.items():
    mapping[animal] = idx

print(mapping)
```

And this will hit `PERF403` asking to use dict comprehension where we do

```python
for animal, idx in other_mapping.items():
    mapping[animal] = idx
```

The only dict comprehension I can imagine is 

```python
mapping = {animal: idx for animal, idx in other_mapping.items()}
```

Which will not do what we want.

Using

```python
mapping.update(other_mapping)
```

Will work instead. But then in this case isn't the `PERF403` hit here a false positive?

**Comment by charliermarsh** [comment]
I think the rule generally wants you to do `mapping.update({animal: idx for animal, idx in other_mapping.items()})` here. We should probably have a separate rule that detects that `{animal: idx for animal, idx in other_mapping.items()}` is an unnecessary comprehension though, and then further corrects to `mapping.update(other_mapping)`.

**Comment by charliermarsh** [comment]
It looks like C416 will further correct that to:

```python
mapping.update(dict(other_mapping.items()))
```

Which is close...

**Comment by charliermarsh** [comment]
Closing for now as I think `mapping.update` is the intended solution.
